File: Perl/approximate-pi.pl
# NEED:
# Perl compiler installed
# USAGE:
# open terminal in this directory and type "perl ./approximate-pi.pl" to compile

#!/usr/bin/perl -w
use strict;
use warnings;

my $iterations;
my $total; 
my $inCircle;
my $radius;
my $input;
my $x;
my $y;
my $pi;

$iterations = 10000;
$total = 0.0;
$inCircle = 0.0;
$radius = 1.0;
# get user input, otherwise use default value
print "Enter number of iterations: ";
$input = <>;
chomp($input);
if(int($input)){
  $iterations = $input;
}
# for-loop that generates random point and checks if they are in circle or not
my @a = (1..$iterations);
for(@a){
  $x = rand(1)*$radius;
  $y = rand(1)*$radius;
  $total++;
  my $distance = sqrt($x**2 + $y**2);
  if($distance < $radius){
      $inCircle++;
  }
}
# pi approximation
$pi = 4 * ($inCircle / $total);
# write output
print "Approximation of pi is: ".$pi;

File: Fortran/approximate-pi.f90
! USAGE
! Install gfortran and MinGW
! Compile code with "gfortran approximate-pi.f90"
! Execute code with "./a.exe"

program approximatePi
  implicit none
  real :: x, y, distance, radius, pi, inCircle
  integer :: i, iterations
  character(*), parameter :: fmt = "(a,g0)"
  write(*, fmt) "Iterations: "
  read(*,*) iterations

  radius = 1.0
  inCircle = 0

  call random_seed()
  do i = 1, iterations 
      call random_number(x)
      call random_number(y)
      distance = sqrt(x**2 + y**2)
      if(distance < radius) inCircle = inCircle + 1
  end do
  pi = 4 * (inCircle / iterations)
  write(*, fmt) "Pi approximation: ", pi
 end